src/server/index.js

```diff
@@ -205,11 +174,15 @@ class KiteServer extends Emitter {
   }
 
   onConnection(ws) {
-    const proto = dnodeProtocol(this.api)
+    const proto = dnodeProtocol(this.api.methods)
     proto.on('request', this.lazyBound('handleRequest', ws))
 
     const id = ws.getId()
-    ws.on('message', this.lazyBound('handleMessage', proto))
+
+    ws.on('message', message => {
+      handleIncomingMessage.call(this, proto, parse(message))
```

Comment:
we're parsing the request in here first and sending to `handleIncomingMessage` as `object` which then we're re-stringifying it in the first line of `handleIncomingMessage`. This can be a requirement but it's better to update `handleIncomingMessage` to accept both version at the same instead of making the expensive call each time.